In [10]:
import pandas as pd
import networkx as nx
import matplotlib.pyplot as plt

# Network Construction 

## 1. Transition Matrix

### 1.0 Import

Read out transition matrix as a Pandas dataframe. No need to use Dask from here because our transition matrix is 1000 x 1000 matrix. 

In [4]:
df = pd.read_csv("Transition_Matrix.csv", index_col=0)

In [5]:
df

11-1011.00  11-1011.03  11-1021.00  11-1031.00  11-2011.00  \
11-1011.00    153583.0        25.0     44281.0       178.0       867.0   
11-1011.03        26.0        47.0        23.0         1.0         0.0   
11-1021.00     61170.0        24.0    319037.0        71.0      1101.0   
11-1031.00       166.0         0.0        49.0        18.0         2.0   
11-2011.00      1806.0         1.0      1620.0         2.0      4329.0   
...                ...         ...         ...         ...         ...   
55-3014.00        17.0         0.0        45.0         0.0         0.0   
55-3015.00        12.0         0.0        37.0         0.0         0.0   
55-3016.00       276.0         0.0       777.0         3.0         3.0   
55-3018.00         0.0         0.0         1.0         0.0         0.0   
55-3019.00        61.0         0.0       116.0         0.0         1.0   

            11-2021.00  11-2022.00  11-2032.00  11-2033.00  11-3012.00  ...  \
11-1011.00     18447.0     23859.0     11130.0        93.0      1159.0  ...   
11-1011.03        30.0        10.0        30.0         0.0         0.0  ...   
11-1021.00     46666.0     76036.0      6822.0        73.0      3190.0  ...   
11-1031.00        22.0        43.0        64.0         0.0         5.0  ...   
11-2011.00      7946.0      3817.0      1716.0        12.0        56.0  ...   
...                ...         ...         ...         ...         ...  ...   
55-3014.00         7.0         8.0         2.0         0.0         1.0  ...   
55-3015.00        10.0        15.0         4.0         0.0         3.0  ...   
55-3016.00       143.0       296.0        46.0         0.0        25.0  ...   
55-3018.00         0.0         0.0         0.0         0.0         0.0  ...   
55-3019.00        42.0        41.0         7.0         0.0         3.0  ...   

            55-2012.00  55-2013.00  55-3011.00  55-3012.00  55-3013.00  \
11-1011.00         6.0         1.0         0.0         0.0         0.0   
11-1011.03         0.0         0.0         0.0         0.0         0.0   
11-1021.00        11.0         2.0         0.0         0.0         0.0   
11-1031.00         0.0         0.0         0.0         0.0         0.0   
11-2011.00         0.0         0.0         0.0         0.0         0.0   
...                ...         ...         ...         ...         ...   
55-3014.00         3.0         0.0         0.0         0.0         0.0   
55-3015.00         1.0         0.0         0.0         0.0         0.0   
55-3016.00         7.0         0.0         0.0         0.0         0.0   
55-3018.00         0.0         0.0         0.0         0.0         0.0   
55-3019.00         7.0         0.0         0.0         0.0         0.0   

            55-3014.00  55-3015.00  55-3016.00  55-3018.00  55-3019.00  
11-1011.00         8.0         9.0        34.0         0.0        24.0  
11-1011.03         0.0         0.0         0.0         0.0         0.0  
11-1021.00        14.0        18.0       116.0         0.0        43.0  
11-1031.00         0.0         0.0         0.0         0.0         0.0  
11-2011.00         0.0         2.0         1.0         0.0         4.0  
...                ...         ...         ...         ...         ...  
55-3014.00        27.0         3.0        17.0         0.0        19.0  
55-3015.00         0.0        29.0         0.0         0.0         1.0  
55-3016.00        31.0        11.0      1119.0         0.0        55.0  
55-3018.00         0.0         0.0         0.0         0.0         0.0  
55-3019.00         2.0         4.0        26.0         0.0       119.0  

[1016 rows x 1016 columns]

### 1.1 Intrepreating our Transition Matrix

In [7]:
total_hops = df.sum().sum()

In [8]:
total_hops

39756794.0

Note that our transition matrix is based on 12,815,753 unique profiles with 51,155,844 rows of job experiences (Refer to Dask_Turtorial.ipynb secion 1.3).

With our hop counting algorithm, we count a total of 39,756,794 hops. 

In [9]:
51155844 - 12815753

38340091

38,340,091 would be the total number of hops across occupations had we only counted experiences sequentially. However, we also count hops where one starts a new job occurs during an existing role. Thus, we may count more than one hop from each job experience. 

## 2. Occupation-Specific Networks